#161967126 User email verification upon registration

Bryan-Cee opened the pull request:
**What does this PR do?**

Enable email verification upon registration.

**Description of tasks to be completed**

- Add email verification upon registration.
- Fixed errors raised by Flake8
- Add tests for email verification
- Refactor failing test

**How should this be manually tested?**

Clone the repo from your terminal: `$ git clone https://github.com/andela/ah-code-titans.git
`
Create a virtual environment: `$ python3 -m venv <name_of_environment>`

Activate the virtual environment: `$ source <name_of_environment>/bin/activate`

Install project dependencies: `$ pip install -r requirements.txt`

Add the following environment variables and populate them with your settings:

-`$ export DB_NAME='choose_a_dbname'`
-`$ export DB_USER='db_username'`
-`$ export DB_PASSWORD='db_role_password`'
-`$ export DB_HOST='localhost or preffered host'`
-`$ export DB_PORT='server_port_number'`
-`$ export SECRET_KEY='your_secret_key'`

Switch to the appropriate branch: `$ git checkout ft-registration-email-verification-161967126`
Run migrations: `$ python manage.py migrate`
Run tests: `$ python manage.py test`

What are the relevant Pivotal Tracker stories?
[#161967126](https://www.pivotaltracker.com/story/show/161967126)

Checklist

- [x]   My code follows the style guidelines of this project 
- [x]   I have performed a self-review of my own code
- [x]   I have commented my code, particularly in hard-to-understand areas
- [x]   I have made corresponding changes to the documentation
- [x]   My changes generate no new warnings

Screenshots
<img width="661" alt="screenshot 2018-12-06 at 10 04 19" src="https://user-images.githubusercontent.com/19145546/49567314-5be7b300-f93e-11e8-8dce-28269f0b2ecb.png">
<img width="964" alt="screenshot 2018-12-07 at 11 36 34" src="https://user-images.githubusercontent.com/19145546/49636698-64aab880-fa14-11e8-807e-4e166c2a699e.png">

jillo-abdullahi reviewed (changes requested):
`test_login_with_inactive_account` is failing locally. There could be an env variable you have not added here.

jillo-abdullahi commented:
@Bryan-Cee You can rebase to develop to pick up @davishooly's changes and try re-running your tests.

Bryan-Cee committed:
feat(feature): add email verification

 - Add email verification on registration
 - Add tests for email verification
 - Fixed errors from flake8

[Delivers #161967126]

denomash commented on authors/apps/authentication/tests/test_login.py:
```diff
@@ -59,15 +59,14 @@ def test_login_with_empty_password_field(self):
 
-    def test_login(self):
+    def test_login_with_inactive_account(self):
         """ test login success """
```
update this comment to match the function of the test

Darius-Ndubi reviewed (approved):
👍

denomash reviewed (approved):
👍

bettblake08 commented on authors/apps/authentication/tests/test_register.py:
```diff
@@ -16,6 +19,27 @@ def register_user(self, data):
+
+        response_details = self.register_user(self.new_user)
+        user_details = User.objects.get(username=self.new_user['user']['username'])
+        from django.utils.encoding import force_bytes
```
Please move the import to the top.

mutoromike reviewed (changes requested):
Kindly add the register screenshot.

bettblake08 reviewed (commented):
Response message should inform the user check their email.

bettblake08 reviewed (approved):
👍 Looks good!
